### PR TITLE
Upgrade dependencies when updating Golem

### PR DESCRIPTION
**Pull request by banasiakadam60**
Signed-off-by: Adam Banasiak <banasiakadam60@gmail.com>

### CODE REVIEW

**Comment by mfranciszkiewicz** [comment]
The installation script gets stuck on dependency installation. I type the password for sudo and then nothing happens.

Also, docker installation script warns:
```
If you already have Docker installed, this script can cause trouble, which is
why we're displaying this warning and provide the opportunity to cancel the
installation.
```
so we should probably skip updating Docker.

**Review by mfranciszkiewicz — DISMISSED** [review]
Please address the issues from my previous comment

**Comment by banasiakadam60** [comment]
@mfranciszkiewicz 
OK, `https://get.docker.com` does
```
$sh_c "mkdir -p /etc/apt/sources.list.d"
$sh_c "echo deb \[arch=$(dpkg --print-architecture)\] ${apt_url}/repo ${lsb_dist}-${dist_version} ${repo} > /etc/apt/sources.list.d/docker.list"
$sh_c 'sleep 3; apt-get update; apt-get install -y -q docker-engine'
```
So I'll add `docker-engine` to this line https://github.com/golemfactory/golem/blob/update_installer/Installer/Installer_Linux/install.sh#L149

File: Installer/Installer_Linux/install.sh [hunk 1]
```diff
@@ -173,18 +175,18 @@ function install_golem()
 {
     download_package
     result=$?
-    if [[ $result -eq 1 ]]; then
+    if [[ ${result} -eq 1 ]]; then
         return 1
     fi
 
-    tar -zxvf /tmp/$PACKAGE
```
Comment (paulperegud) [inline Installer/Installer_Linux/install.sh:180]: @banasiakadam60 Next time please split changes 'that makes things better' from core of your change. Otherwise commits are filled with noise and harder than necessary to read. In this particular case changing the way local variables are expanded should be made in different commit from change of the way dependencies are upgraded.

**Review by paulperegud — APPROVED** [review]
LGTM (although I haven't managed to test it properly yet)

**Comment by banasiakadam60** [comment]
#1124

**Comment by banasiakadam60** [comment]
@paulperegud It would be great if you could test it tomorrow. Eg. launch script, downgrade `docker` and `geth` (`ethereum` and `docker-engine` packages) and rerun this script.

**Comment by banasiakadam60** [comment]
Tested, it works for me
